python/cudf/dataframe/dataframe.py

```diff
@@ -246,10 +246,23 @@ def __getitem__(self, arg):
                 for col in arg:
                     df[col] = self[col]
             return df
+        elif isinstance(arg, DataFrame):
+            return self.mask(arg)
         else:
             msg = "__getitem__ on type {!r} is not supported"
             raise TypeError(msg.format(type(arg)))
 
+    def mask(self, other):
+        df = self.copy()
+        for col in self.columns:
+            if col in other.columns:
+                boolbits = cudautils.compact_mask_bytes(
+                           other[col].to_gpu_array())
+            else:
+                boolbits = cudautils.full(len(self[col]), 0, np.uint8)
```

Comment:
You still need to compact this otherwise this mask is way too big. It would be more efficient to create the mask of the correct size and fill that with zeros though using `cudautils.make_empty_mask(len(self[col]))`

Reply:
```suggestion
                boolbits = cudautils.make_empty_mask(len(self[col]))
```